crates/celo-revm/src/handler.rs

```diff
@@ -650,6 +694,31 @@ where
                 return Err(ERROR::from(OpTransactionError::HaltedDepositPostRegolith));
             }
         }
+
+        // Merge system call logs with main transaction logs
+        let chain = evm.ctx().chain();
+        let has_system_logs = !chain.cip64_actual_tx_system_call_logs_pre.is_empty()
+            || !chain.cip64_actual_tx_system_call_logs_post.is_empty();
+
+        if has_system_logs {
+            match &mut result.result {
+                ExecutionResult::Success { logs, .. } => {
+                    // Build merged logs: pre_logs + main_tx_logs + post_logs
+                    let mut merged_logs = chain.cip64_actual_tx_system_call_logs_pre.clone();
+                    merged_logs.append(logs);
```

Comment:
Does it make sense to not consume the logs in result and copy it?
```suggestion
                    merged_logs.extend(logs.clone());
```

Reply:
Maybe the `result` which includes `logs` is not cleared, just returned from the `output` function and reused later